2.13 release notes (#8445)
grafana/mimir: docs/sources/mimir/release-notes/v2.13.md

---
title: "Grafana Mimir version 2.13 release notes"
menuTitle: "V2.13 release notes"
description: "Release notes for Grafana Mimir version 2.13"
weight: 1100
---

# Grafana Mimir version 2.13 release notes

<!-- vale Grafana.GoogleWill = NO -->
<!-- vale Grafana.Timeless = NO -->
<!-- Release notes are often future focused -->

Grafana Labs is excited to announce version 2.13 of Grafana Mimir.

The highlights that follow include the top features, enhancements, and bug fixes in this release.
For the complete list of changes, refer to the [CHANGELOG](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

## Features and enhancements

- **Improved CPU performance in processing queries with regular expressions** that contain many alternations (for example `foo|bar|baz|...`).

- **Improved OTLP ingestion performance** by using native translation, which reduces memory usage in the distributor by up to 30% and CPU usage by up to 8%.

> - Configurable S3 bucket lookup type improves interoperability with **S3-compatible providers such as Tencent and Alibaba**. Configure lookup type via `-<prefix>.s3.bucket-lookup-type` or `-common.storage.s3-bucket-lookup-type`.

Reviewer: ```suggestion
- Configurable S3 bucket lookup type improves interoperability with **S3-compatible providers, such as Tencent and Alibaba**. Configure lookup type via `-<prefix>.s3.bucket-lookup-type` or `-common.storage.s3-bucket-lookup-type`.
```

- **Configuration of TLS for S3 buckets** is now possible via `-<prefix>.s3.http.*` and `-common.storage.s3.http.*` configuration options.

- **Mimirtool can now verify the validity of a Mimir runtime configuration** file with the `mimirtool runtime-config verify` command.

> - **Active series are now updated along with owned series.** This means the number of active series for a tenant is much more accurate after scaling out ingesters. As a result ingesters now enforce more precisely tenants' series limits. This is only enabled when `-ingester.track-ingester-owned-series` or `-ingester.use-ingester-owned-series-for-limits` are enabled.

Reviewer: ```suggestion
- **Active series are now updated along with owned series.** This means the number of active series for a tenant is more accurate after scaling out ingesters. As a result, ingesters now more precisely enforce tenants' series limits. This feature is only enabled when `-ingester.track-ingester-owned-series` or `-ingester.use-ingester-owned-series-for-limits` are enabled.
```

> - **Store-gateway can be configured to explicitly disable or enable tenants**. This is useful in cases when compaction slows down for a tenant to temporarily exclude a tenant while compaction catches up without impact other tenants.

Reviewer: ```suggestion
- **Store-gateway can be configured to explicitly disable or enable tenants**. This is useful in cases where compaction slows down for a tenant to temporarily exclude another tenant while compaction catches up without affecting other tenants.
```

- **Remote read (`/prometheus/api/v1/read`) is becoming a first-class endpoint in Mimir** with support in `query stats` logs and query blocking in the query-frontend. More coming in future releases.

Additionally, the following previously experimental features are now considered stable:

- Rules tenant federation via the `source_tenants` field in rule groups.

> - Enabling and recording and alerting rules evaluation on a per-tenant basis.

Reviewer: ```suggestion
- Enabling, recording, and alerting rules evaluation on a per-tenant basis.
```

Author: "Enabling" refers to "recording and alerting rules", so I'll drop the first comma

```suggestion
- Enabling recording, and alerting rules evaluation on a per-tenant basis.
```

- Limiting the number of tenants a federated query can query.

## Important changes

In Grafana Mimir 2.13 the following behavior has changed:

- The default Docker image `grafana/mimir` is now based on the distroless image `gcr.io/distroless/static-debian12`.
  See [Debugging distroless container images](https://grafana.com/docs/mimir/next/manage/mimir-runbooks/#debugging-distroless-container-images-in-kubernetes) for more details on how to
  work with distroless images.

> - Error logs in the ingester are now sampled at 10% by default. Sampled log lines will contain `(sampled 1/10)`. All the discarded samples will still be tracked by the `cortex_discarded_samples_total` metric.

Reviewer: ```suggestion
- Error logs in the ingester are now sampled at 10% by default. Sampled log lines contain `(sampled 1/10)`. The `cortex_discarded_samples_total` metric still tracks all discarded samples.
```

Reviewer: Small update to avoid "will" and to use active voice, per https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/.

---
title: "Grafana Mimir version 2.13 release notes"
menuTitle: "V2.13 release notes"
description: "Release notes for Grafana Mimir version 2.13"
weight: 1100
---

# Grafana Mimir version 2.13 release notes

<!-- vale Grafana.GoogleWill = NO -->
<!-- vale Grafana.Timeless = NO -->
<!-- Release notes are often future focused -->

Grafana Labs is excited to announce version 2.13 of Grafana Mimir.

The highlights that follow include the top features, enhancements, and bug fixes in this release.
For the complete list of changes, refer to the [CHANGELOG](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).

## Features and enhancements

> - **Store-gateway can be configured to explicitly disable or enable tenants**. This is useful in cases when compaction slows down for a tenant to temporarily exclude a tenant while compaction catches up without impact other tenants.

Reviewer 2: Is this the most exciting new feature? I'd expect us to put the top highlights first.

Author: haha, good point. This was largely ordered as the changelog

Reviewer 2: Looks reasonable to me, @zhehao-grafana might have opinions though.

> - Configurable S3 bucket lookup type improves interoperability with **S3-compatible providers such as **Tencent and Alibaba\*\*\*\*. Configure lookup type via `-<prefix>.s3.bucket-lookup-type` or `-common.storage.s3-bucket-lookup-type`.

Reviewer 2: Formatting here with the `*`s looks off - should the whole of `S3-compatible providers such as Tencent and Alibaba` be bold?
```suggestion
- Configurable S3 bucket lookup type improves interoperability with **S3-compatible providers such as Tencent and Alibaba**. Configure lookup type via `-<prefix>.s3.bucket-lookup-type` or `-common.storage.s3-bucket-lookup-type`.
```

- **Mimirtool can now verify the validity of a Mimir runtime configuration** file with the `mimirtool runtime-config verify` command.

- **Configuration of TLS for S3 buckets** is now possible via `-<prefix>.s3.http.*` and `-common.storage.s3.http.*` configuration options.

- **Improved OTLP ingestion performance** by using native translation, which reduces memory usage in the distributor by up to 30% and CPU usage by up to 8%.

> - **Active series are now updated along with owned series.** This means the number of active series for a tenant is much more accurate after adding ingesters.

Reviewer 2: Might be worth explaining more why this is useful to a user / operator - doesn't this mean that there should be fewer issues with limits during ingester scaling, for example?

Author: yeah, fair, how is https://github.com/grafana/mimir/pull/8445/commits/c29fbf4fdf861a1b440f4d8edd54a59861afd260?

Reviewer 2: LGTM, thanks.

- **Remote read (`/prometheus/api/v1/read`) is becoming a first-class endpoint in Mimir** with support in `query stats` logs and query blocking in the query-frontend. More coming in future releases.

- **Improved CPU performance in processing queries with regular expressions** that contain many alternations (for example `foo|bar|baz|...`).

Additionally, the following previously experimental features are now considered stable:

- Rules tenant federation via the `source_tenants` field in rule groups.
- Enabling and recording and alerting rules evaluation on a per-tenant basis.
- Limiting the number of tenants a federated query can query.

## Important changes

In Grafana Mimir 2.13 the following behavior has changed:

- The default Docker image `grafana/mimir` is now based on the distroless image `gcr.io/distroless/static-debian12`.

>   See [Debugging distroless container images](https://grafana.com/docs/mimir/next/manage/mimir-runbooks/#debugging-distroless-container-images-in-kubernetes) for more details on how to

Reviewer 2: Should this point to a 2.13-specific URL, rather than `next`?

Author: i wasn't sure what we want. `next` is probably not it. `latest` is more likely to give accurate results in future releases, but is also more likely to break as pages are reordered. `v2.13.x` won't break, but the information can become outdated. 

@tacole02 can you take a look here? This is a link to a section only published with the 2.13 release. We aren't sure what versioned link to use for it.

Reviewer: Will the section be included in future releases that follow 2.13? If yes, then I think we should go with `latest`.

If this is only for 2.13 and won't go into future releases, we should specify 2.13 in the URL.

Author: Thanks! it will be included in future releases, so I'll just use `latest`

```suggestion
  See [Debugging distroless container images](https://grafana.com/docs/mimir/latest/manage/mimir-runbooks/#debugging-distroless-container-images-in-kubernetes) for more details on how to
```

  work with distroless images.

> - Errors logs in the ingester are now sampled at 10% by default. Sampled log lines will contain `(sampled 1/10)`. All the discarded samples will still be tracked by the `cortex_discarded_samples_total` metric.

Reviewer 2: ```suggestion
- Error logs in the ingester are now sampled at 10% by default. Sampled log lines will contain `(sampled 1/10)`. All the discarded samples will still be tracked by the `cortex_discarded_samples_total` metric.
```

- Anonymous usage statistics now include actual CPU usage instead of available CPU cores.

> - Continuous-test is no longer a standalone binary and is now part of Mimir as its own target. The published Docker images has been updated to use the new packaging.

Reviewer 2: ```suggestion
- Continuous-test is no longer a standalone binary and is now part of Mimir as its own target. The published Docker image has been updated to use the new packaging.
```

The following deprecated configuration options are removed in Grafana Mimir 2.13:

- The configuration option `-log.buffered`, which was deprecated in 2.11 and is now enabled by default.

## Experimental features

Grafana Mimir 2.13 includes new features that are considered experimental and disabled by default.
Use them with caution and report any issues you encounter:

> - **New PromQL query engine** can be configured via the `-querier.promql-engine=mimir` CLI flag on queriers.
>   It aims to provide better utilization than the upstream Prometheus PromQL engine when processing large sets of data.

Author: fair point. Let's keep the fanfare for next releases
```suggestion
```

> - Support for [**HAProxy PROXY protocol**](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) can now be enabled via `-server.proxy-protocol-enabled=true`.

Reviewer 3: Can we add some context for this? I have no idea what this means.

Reviewer 4: My understanding is that PROXY protocol is a way to communicate information about client at the beginning of TCP connection. Our HTTP and gRPC servers support this to know who they communicate with, basically replacing information from TCP header.

Author: at a second look I don't think we should even be including this in the release notes. Mimir doesn't do anything with the client's address, so enabling this doesn't bring anything for Mimir. I found these docs in dskit which were added along with the feature

https://github.com/grafana/dskit/blob/8569e93a110e8b5642be030dbebccab5f6a41bad/server/PROXYPROTOCOL.md

If you agree, I'll just remove this line from the notes

Reviewer 3: Yeah, I think that makes sense. This doesn't seem relevant for Mimir.

Reviewer 4: Agree, don't see much benefit for Mimir users.

Author: ```suggestion
```

> Unless they run Mimir directly behind some proxy that supports this protocol.

that looks niche. We still have the changelog entry
